### PR TITLE
Overlays uuid

### PR DESCRIPTION
**Pull request by sethalves**
- overlays are now identified by UUIDs rather than unsigned integers
- overlays can be parents (in the SpatiallyNestable sense)
- Overlays::findRayIntersection arguments now match EntityScriptingInterface::findRayIntersection

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- make sure overlays still work
- use edit.js and use the affordances
- run the dice script and make sure the buttons still work

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2b02998fae4cb9ecfb05128c5a4962698d9b05dd available at: https://gist.github.com/b13b86622eeb038534181fa496c37a11<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-2b02998.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-2b02998.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** FAIL
**Additional Test Results + Observations**:  Tested in ```dev-playa```.  Overlays works in general except for the 'X' on the tablet which when clicking on it with the stylus or Oculus Touch, had no effect.  In addition, ran ```dice``` and the dice rezzing overlay worked as it rezzed dice however the + and - had no effect.  I don't know if this is an issue with the overlays within their respective code functionality or a failure of this PR however it does not function as expected.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47cdade1e9a3ba3d123678a46c3a9552beb386f1 available at: https://gist.github.com/f65f01bf0370bb0630cbe7b8d0bba5fa<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-47cdade.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-47cdade.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: def669364b734f73be71ea4097278b6a9d01c4ba available at: https://gist.github.com/f65f01bf0370bb0630cbe7b8d0bba5fa<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-def6693.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-def6693.dmg

**Comment by kevinmthomasse** [comment]
@sethalves I will retest as I saw you made the fix for the home button however I wanted to confirm with you the dice script issue as the x and - have no effect is that expected behavior?

**Comment by sethalves** [comment]
yeah I haven't figured out what's wrong with the dice buttons.  There's still something wrong.

**Comment by kevinmthomasse** [comment]
@sethalves ok to retest this ignoring the dice script and testing only for the tablet fix?

**Comment by sethalves** [comment]
nah, hold off until I get another stab at it.  My fear is that the dice buttons' failure to work is hiding some deeper problem.

**Comment by sethalves** [comment]
@kevinmthomasse okay, maybe this works now.  Give it another try, when you can.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4a650ff2bb81efcdea2c8ff33a19396085266816 available at: https://gist.github.com/f65f01bf0370bb0630cbe7b8d0bba5fa<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-4a650ff.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-4a650ff.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 70c98531890f4fec7494298408e5dd3afe1bc6ce available at: https://gist.github.com/f65f01bf0370bb0630cbe7b8d0bba5fa<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-70c9853.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-70c9853.dmg

**Comment by kevinmthomasse** [comment]
re-testing

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -6891,3 +6894,13 @@ void Application::toggleMuteAudio() {
     auto menu = Menu::getInstance();
     menu->setIsOptionChecked(MenuOption::MuteAudio, !menu->isOptionChecked(MenuOption::MuteAudio));
 }
+
+OverlayID Application::getTabletScreenID() {
+    auto HMD = DependencyManager::get<HMDScriptingInterface>();
+	return HMD->getCurrentTabletScreenID();
```
Comment (hyperlogic) [inline interface/src/Application.cpp]: indenting is wacked

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -6891,3 +6894,13 @@ void Application::toggleMuteAudio() {
...
+}
+
+OverlayID Application::getTabletHomeButtonID() {
+    auto HMD = DependencyManager::get<HMDScriptingInterface>();
+	return HMD->getCurrentHomeButtonUUID();
```
Comment (hyperlogic) [inline interface/src/Application.cpp]: 4 space indents, please

File: interface/src/Application.h [hunk 3]
```diff
@@ -298,6 +298,9 @@ class Application : public QApplication,
     Q_INVOKABLE void sendHoverOverEntity(QUuid id, PointerEvent event);
     Q_INVOKABLE void sendHoverLeaveEntity(QUuid id, PointerEvent event);
 
+    OverlayID getTabletScreenID();
+    OverlayID getTabletHomeButtonID();
```
Comment (hyperlogic) [inline interface/src/Application.h]: can you make these methods const?

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**:  Tested in ```dev-welcome```.  Overlays works correctly and tablet X works as expected as well.

File: interface/src/ui/overlays/Overlays.cpp [hunk 4]
```diff
@@ -441,23 +449,23 @@ RayToOverlayIntersectionResult Overlays::findRayIntersection(const PickRay& ray)
...
     extraInfo()
-{ 
+{
 }
 
 QScriptValue RayToOverlayIntersectionResultToScriptValue(QScriptEngine* engine, const RayToOverlayIntersectionResult& value) {
     auto obj = engine->newObject();
     obj.setProperty("intersects", value.intersects);
-    obj.setProperty("overlayID", value.overlayID);
+	obj.setProperty("overlayID", OverlayIDtoScriptValue(engine, value.overlayID));
```
Comment (hyperlogic) [inline interface/src/ui/overlays/Overlays.cpp]: indenting

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3b181eb80378db12d1917efa7cc20040fa15cc60 available at: https://gist.github.com/f65f01bf0370bb0630cbe7b8d0bba5fa<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-3b181eb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9699-3b181eb.dmg

**Comment by BillWindwalker** [comment]
I talked about this back when I was in second life but Second Life didn't
have such things

On Feb 17, 2017 11:50 AM, "Seth Alves" <notifications@github.com> wrote:


   - overlays are now identified by UUIDs rather than unsigned integers
   - overlays can be parents (in the SpatiallyNestable sense)
   - Overlays::findRayIntersection arguments now match
   EntityScriptingInterface::findRayIntersection

------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/highfidelity/hifi/pull/9699
Commit Summary

   - working toward using uuids for overlay IDs
   - uuid keys for overlays appears to work
   - Merge branch 'master' of github.com:highfidelity/hifi into
   overlays-uuid
   - allow overlays to be parents
   - hook up overlaysToDiscard and overlaysToInclude in
   Overlays::findRayIntersection

File Changes

   - *M* interface/src/Application.cpp
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-0> (21)
   - *M* interface/src/Application.h
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-1> (6)
   - *M* interface/src/InterfaceParentFinder.cpp
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-2> (10)
   - *M* interface/src/devices/DdeFaceTracker.cpp
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-3> (2)
   - *M* interface/src/devices/DdeFaceTracker.h
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-4> (2)
   - *M* interface/src/ui/overlays/Base3DOverlay.h
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-5> (3)
   - *M* interface/src/ui/overlays/Overlay.cpp
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-6> (22)
   - *M* interface/src/ui/overlays/Overlay.h
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-7> (26)
   - *M* interface/src/ui/overlays/OverlayPanel.cpp
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-8> (6)
   - *M* interface/src/ui/overlays/OverlayPanel.h
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-9> (11)
   - *M* interface/src/ui/overlays/Overlays.cpp
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-10> (131)
   - *M* interface/src/ui/overlays/Overlays.h
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-11> (109)
   - *M* interface/src/ui/overlays/Web3DOverlay.cpp
   <https://github.com/highfidelity/hifi/pull/9699/files#diff-12> (4)

Patch Links:

   - https://github.com/highfidelity/hifi/pull/9699.patch
   - https://github.com/highfidelity/hifi/pull/9699.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/highfidelity/hifi/pull/9699>, or mute the thread

**Comment by BillWindwalker** [comment]
Cc

**Pull request merged**